packages/@pollyjs/adapter-node-http/src/-private/transport-wrapper.js

```diff
@@ -138,17 +138,16 @@ export default class TransportWrapper {
       ...NodeUrl.parse(pollyRequest.url)
     });
 
-    request.on('error', error => {
-      throw error;
+    const requestPromise = new Promise((resolve, reject) => {
+      request.once('response', response => resolve(response));
+      request.once('error', reject);
     });
 
     // Write the request body
     chunks.forEach(chunk => request.write(chunk));
 
-    const response = await new Promise(resolve => {
-      request.once('response', response => resolve(response));
-      request.end();
-    });
+    request.end();
```

Comment:
What was the need to move this out of the above promise?

Reply:
The request is ended prematurely, because the call to end happens before the request body has been written. Luckily this caused a test to fail :)